features/events/daemon/src/main/java/org/opennms/netmgt/eventd/sink/EventsModule.java

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2016 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2016 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+/**
+ * @author ms043660 (Malatesh.Sudarshan@cerner.com)
+ */
+package org.opennms.netmgt.eventd.sink;
+
+import java.util.Objects;
+
+import org.opennms.core.ipc.sink.api.AggregationPolicy;
+import org.opennms.core.ipc.sink.api.AsyncPolicy;
+import org.opennms.core.ipc.sink.xml.AbstractXmlSinkModule;
+import org.opennms.netmgt.config.api.EventdConfig;
+import org.opennms.netmgt.events.api.EventsWrapper;
+import org.opennms.netmgt.xml.event.Log;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EventsModule extends AbstractXmlSinkModule<EventsWrapper, Log> {
+
+	public static final String MODULE_ID = "Events";
+
+	private static final Logger LOG = LoggerFactory.getLogger(EventsModule.class);
+
+	private final EventdConfig m_config;
+
+	public EventsModule(EventdConfig config) {
+		super(Log.class);
+		this.m_config = config;
+	}
+
+	@Override
+	public String getId() {
+		return MODULE_ID;
+	}
+
+	@Override
+	public int getNumConsumerThreads() {
+		return m_config.getNumThreads();
+	}
+
+	@Override
+	public AggregationPolicy<EventsWrapper, Log, Log> getAggregationPolicy() {
+		return new AggregationPolicy<EventsWrapper, Log, Log>() {
+
+			@Override
+			public int getCompletionSize() {
+				return m_config.getBatchSize();
+			}
+
+			@Override
+			public int getCompletionIntervalMs() {
+				return m_config.getBatchIntervalMs();
+			}
+
+			@Override
+			public Object key(EventsWrapper message) {
+				return message.getEvents();
+			}
+
+			@Override
+			public Log aggregate(Log accumulator, EventsWrapper eventsWrapper) {
+				accumulator = eventsWrapper.getEvents();
```

Comment:
The  aggregate method should add the given object to the existing accumulator, and create a new one if necessary. Here we are ignoring the existing accumulator and just using the log in the message.

The `EventsWrapper`  probably be changed to wrap a single event, and this message should updated to roll those up into the `Log`.

Reply:
this has been addressed.